File: src/main/scala/ejercicios/ejercicio1.scala
package ejercicios

import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._

object ejercicio1 {

  def main(args: Array[String]): Unit = {
    val spark = SparkSession
      .builder
      .appName("LibroSpark")
      .master("local[2]")
      .getOrCreate()

    //añado en el pom la dependencia spark-avro correspondiente a la versión de Spark 3.3.2 y leo los archivos
    val productDf = spark.read.format("avro").load("src/main/resources/retail_db/products_avro/part-m-00000.avro")
    productDf.show(10, truncate = false)
    productDf.printSchema()

    //filtro según los criterios del ejercicio
    val solucionDf = productDf
      .filter(
        col("product_price").between(20,23)
          .and(col("product_name").like("Nike%"))
      )
      .orderBy(col("product_price"))
    solucionDf.show(30, truncate = false)

    //guardo en el formato especificado
    solucionDf.write.format("parquet")
      .mode("overwrite")
      .option("compression", "gzip")
      .save("src/main/dataset/q1/solution")


  }

}

File: src/main/python/ejercicio1.py
from pyspark.sql import SparkSession
from pyspark.sql.types import *
from pyspark.sql.functions import *

spark = SparkSession \
    .builder \
    .appName("pruebaSpark") \
    .master("local[2]") \
    .getOrCreate()

# leo los archivos
product_df = spark.read.format("avro").load("C:/Users/julia.blanco/Desktop/repositorios/pruebaSpark/src/main/resources/retail_db/products_avro")
product_df.show(10, truncate=False)
product_df.printSchema()

# filtro según los criterios del ejercicio (se puede usar también .where)
q1_df = product_df.filter(
    (product_df.product_price.between(20,23)) &
    (product_df.product_name.like("Nike%"))
).orderBy(product_df.product_price)
q1_df.show(30, truncate=False)

# guardo en el formato especificado
q1_df.write.format("parquet") \
    .mode("overwrite") \
    .option("header", "true") \
    .option("compression", "gzip") \
    .save("C:/Users/julia.blanco/Desktop/repositorios/pruebaSpark/src/main/dataset/q1/solution")